server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1707,6 +1701,24 @@ public void trimUnreferencedReaders() throws IOException {
             closeOnTragicEvent(ex);
             throw ex;
         }
+        // Do sync outside the writeLock to avoid blocking all writing thread.
+        try {
+            // The checkpoint is used when opening the translog to know which files should be recovered from.
+            // We now update the checkpoint to ignore the file we are going to remove.
+            // Note that there is a provision in recoverFromFiles to allow for the case where we synced the checkpoint
+            // but crashed before we could delete the file.
+            if (!toDeleteReaderList.isEmpty()) {
+                sync();
```

Comment:
We can skip `sync` if the `getLastSyncedCheckpoint().minTranslogGeneration >= minReferencedGen`. Can you benchmark if this optimization improves the translog performance?

Reply:
![1](https://user-images.githubusercontent.com/18375623/64107010-08f73b00-cdac-11e9-8aa0-87ccd396e059.png)

Reply:
Sync operation has been move out of writeLock，so  checking "getLastSyncedCheckpoint().minTranslogGeneration >= minReferencedGen"  before  sync show no significant effect.
![1](https://user-images.githubusercontent.com/18375623/64107289-b23e3100-cdac-11e9-801c-2e1c4ee07b8b.png)